kafka/README.md

```diff
@@ -6,16 +6,36 @@ This document defines how to describe Kafka-specific information on AsyncAPI.
 
 ## Version
 
-Current version is `0.1.0`.
+Current version is `0.2.0`.
 
 
 <a name="server"></a>
 
 ## Server Binding Object
 
-This object MUST NOT contain any properties. Its name is reserved for future use.
+This object contains information about the server representation in Kafka.
 
+##### Fixed Fields
 
+Field Name | Type | Description | Applicability [default] | Constraints
+---|:---:|:---:|:---:|---
+`schemaRegistryUrl` | string (url) | API URL for the Schema Registry used when producing Kafka messages (if a Schema Registry was used) | OPTIONAL | -
+`schemaRegistryVendor` | string | Identifies the Kafka serdes library that should be used (e.g. `confluent`, `apicurio`, or `ibm`) | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified
+`schemaRegistryAvailable` | boolean | Specifies if the Schema Registry identified in `schemaRegistryUrl` is available for use by consumers of the AsyncAPI spec | OPTIONAL [true] | MUST NOT be specified if `schemaRegistryUrl` is not specified
```

Comment:
I have some doubt with `schemaRegistryAvailable` as explained in https://github.com/asyncapi/bindings/issues/41#issuecomment-785898884.

I think we still need a more generic `useSchemaRegistry` flag that is tied to the operation level. For me it brings information to the producer that it should integrate with a registry - and this very early in the lifecycle of the API, long before knowing where it will land in terms of servers.

Reply:
I think there are two issues.

1) Describing the data that is in the message
2) Describing other servers where schema information is stored


**(1) Describing the data that is in the message**

I was trying to address this in the message binding object, using the `schemaIdLocation` value.

If the message was produced with a Confluent serdes, then the payload will have the schema id and version inserted at the start. (schemaIdLocation = payload)

(This means you could consume using a Confluent serdes library, which knows how to handle that. Or you can consume yourself by swallowing the beginning bytes before attempting the Avro deserialization)

If the message was produced in, well.... pretty much any other way, then the payload will just be serialized Avro data. (schemaIdLocation = header or none)

(This means you can consume it with a non-Confluent serdes library if you have access to a schema registry, or you can just Avro deserialize the message contents yourself)  


**(2) Describing servers where schema information is stored**

As this isn't an attribute of the data on the topic, I felt like that the server bindings was the best place to put it. 

As it is information about the schema registry that stores schema data (and whether or not people are allowed to connect to it) and not an attribute of the data on the topic, I didn't think it belonged in the message binding. 

Similarly, with `schemaRegistryAvailable` (or `useSchemaRegistry` or whatever we end up naming it) as it is information about the schema registry, and whether or not people are allowed to connect to it, it's still not an attribute of the data on the topic, I didn't think it belonged in the message binding. 

That leaves putting it in the operation, channel, or server bindings as the most obvious places.

The benefit of putting it in at the server level is that it means you don't have to duplicate it. You declare it once - "here's my schema registry and you can use it" or "here's my schema registry, but sorry it's private and you can't use it". 

The drawback of putting it at the server level is that if you have some topics where you would allow people to use the schema registry, and other topics where you won't, then you'd need to declare them in separate AsyncAPI specs. (_In my opinion, that seems like an unlikely edge case, so that seems okay to me as a workaround_)

The main benefit I can think of for putting it in the operation binding would be if you treated producing and consuming differently - so you could say people are allowed to use a schema registry for sending messages but not receiving them (or vice versa). Are there other benefits I'm missing?

I can't think of a strong reason for putting it in the channel binding. As I say, it seems unusual to say a schema registry is available for use for one schema but not for another. And if that's really the case, then you could address that in separate specs. 

Dunno, to be honest - I don't have a use case in mind for this field, so I'm very much guessing here.

Reply:
I'm totally aligned with use-case (1) 😉 

**Some comments regarding use-case (2)**

Agree with you on the location : either server or operation (not message related and can't see any reason it would be channel)

A concern I have with putting this into `servers` is that I met several customers - that are currently using OpenAPI v3 and are looking at AsyncAPI - that are not using `servers` or are using them very late in the API lifecycle... When architects or domain experts at these customers are designing an API, they indeed do not know (and even do not care) on what servers it will be hosted! And I can easily understand that!

Sometimes, the `server` part is later completed as the API implementation is going through QA and Production - usually when it's finally published onto the Developer Portail of the in-house API Management solution - but it's not even a mandatory step.

Now translating to AsyncAPI and our Kafka concerns, I think the having this information of "will we use a registry ?" is very important at design time - because we may have some extra work designing Avro schema as well - and very valuable for developers to come. Moreover I think it's totally orthogonal with the fact of knowing the registry server (`schemaRegistryUrl`) and the fact of knowing if we are going to make it available (`schemaRegistryAvailable`).

So to summarize things up:
* Totally agree with `schemaRegsitryUrl` and `schemaRegistryAvailable` at the `server` level. The `server` block may be not specified at first and then added later in the lifecycle,
* Still thinking we need `useSchemaRegistry` independently at the `operation` level to tell developers, producers and consumers that there's possibly an extra schema to design / publish / retrieve

NB: Another approach could be to deduce this usage of schema registry from the presence of `schemaIdLocation`. If it has a value that is not `none` then it implies that we're using a schema registry ... I think it should do the trick but for me it's far less explicit.

Reply:
That's useful insight, thanks - as I said, I was guessing because I didn't have a use case in mind, so it's useful to have a more concrete use case as guidance.

I've added `useSchemaRegistry` to the operation binding.

I'm still a little unsure about having both `schemaRegistryAvailable` and `useSchemaRegistry` though - mainly because  a contradictory spec that sets `schemaRegistryAvailable` to `false` and `useSchemaRegistry` to `true` would be telling developers they should use a schema registry, but they're not allowed to connect to it.

What do you think?

Reply:
I've removed it for now, as we can always add it in a future PR.